ApiConnection.cs

```diff
@@ -69,16 +70,23 @@ public void DownloadEmail(Guid emailGuid, string directory)
             var email = SearchFolder("Emails", JObject.FromObject(new { ItemGUID = emailGuid }));
             string emailName = email.Value<JArray>("Data").First().Value<string>("FileAs");
             string emailExtension = email.Value<JArray>("Data").First().Value<string>("EmailFileExtension");
-            string filePath = Path.Combine(directory, $"{RemoveInvalidFileNameChars(emailName)}{emailExtension}");
+            string[] emailGuidParts = email.Value<JArray>("Data").First().Value<string>("ItemGUID").Split('-');
+            string filePath = Path.Combine(directory, $"{FileHelper.RemoveInvalidFileNameChars($"{emailName} ({emailGuidParts[0]})" ?? "INVALID_NAME").Trim()}{emailExtension}");
 
             DownloadFile(emailGuid, filePath, directory);
         }
 
         private void DownloadFile(Guid fileGuid, string filePath, string directory)
         {
-            if (File.Exists(directory))
+            // Avoid hitting Windows max path length limit
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
             {
-                Logger.LogDebug($"File '{directory}' already exist");
+                filePath = FileHelper.TrimPathLength(filePath, 260);
```

Comment:
Lepší by bylo trimovat ten emailName, klidně na něco menšího (třeba 64 znaků?), v těch velkých názvech se asi stejně nikdo nevyznaná, jde o to, aby ti tam pak zůstala i ta unikátní část.

Reply:
Jinak pokud to spouštíš jako admin, tak by ses měl dostat i k Server_ID, což je rostoucí řada (INT).

Reply:
@rstefko fixed, přidal jsem tam trim na file name i folder name, ale tenhle check jsem tam radši ještě nechal, ten script by mohl někdo spustit z nějakého deeply nested adresáře a potom by ten trim na 64 znaků v názvu nestačil. Nechal jsem to tam teda jen jako poslední záchranu.

Taky jsem dodělal přidávání první části GUID do názvu dokumentů, stejně jako to je u emailů, aby nedocházelo k nějakým falešným duplicitám kvůli stejnému názvu.